kratos/includes/initial_state.h

```diff
@@ -0,0 +1,279 @@
+                noalias(mInitialStrainVector) = rImposingEntity;
+            } else if (InitialImposingType == static_cast<int>(InitialImposingType::StressOnly)) {
+                noalias(mInitialStressVector) = rImposingEntity;
+            }
+        }
+
+        // Selective constructor for vectors (E, S)
+        InitialState(const Vector& rInitialStrainVector,
+                     const Vector& rInitialStressVector)
+        {
+            const SizeType voigt_size_1 = rInitialStrainVector.size();
+            const SizeType voigt_size_2 = rInitialStressVector.size();
+            const SizeType dimension = (voigt_size_1 == 6) ? 3 : 2;
+            KRATOS_ERROR_IF(voigt_size_1 <= 0 || voigt_size_2 <= 0) << "The imposed vector is null..." << std::endl;
+
+            mInitialStressVector.resize(voigt_size_1, false);
+            mInitialStrainVector.resize(voigt_size_1, false);
+            mInitialDeformationGradientMatrix.resize(dimension, dimension, false);
+
+            noalias(mInitialDeformationGradientMatrix) = ZeroMatrix(dimension, dimension);
+            noalias(mInitialStressVector) = rInitialStressVector;
+            noalias(mInitialStrainVector) = rInitialStrainVector;
+        }
+
+        // Selective constructor for Deformation Gradient only
+        InitialState(const Matrix& rInitialDeformationGradientMatrix)
+        {
+            const SizeType dimension = rInitialDeformationGradientMatrix.size1();
+            const SizeType voigt_size = (dimension == 3) ? 6 : 3;
+            KRATOS_ERROR_IF(dimension <= 0) << "The imposed Matrix is null..." << std::endl;
+
+            mInitialDeformationGradientMatrix.resize(dimension, dimension, false);
+            noalias(mInitialDeformationGradientMatrix) = rInitialDeformationGradientMatrix;
+
+            mInitialStressVector.resize(voigt_size, false);
+            mInitialStrainVector.resize(voigt_size, false);
+            noalias(mInitialStrainVector) = ZeroVector(voigt_size);
+            noalias(mInitialStressVector) = ZeroVector(voigt_size);
+        }
+
+        /// Destructor.
+        virtual ~InitialState() {}
+
+
+        ///@}
+        ///@name Operations
+        ///@{
+
+        //*********************************************
+        //public API of intrusive_ptr
+        unsigned int use_count() const noexcept
+        {
+            return mReferenceCounter;
+        }
+        //*********************************************
+
+        /**
+         * @brief This method sets the initial strain vector
+         * @param rInitialStrainVector The vector to be set
+         */
+        void SetInitialStrainVector(const Vector& rInitialStrainVector) {
+            const SizeType voigt_size = rInitialStrainVector.size();
+            KRATOS_ERROR_IF(voigt_size <= 0) << "The imposed vector is null..." << std::endl;
+
+            mInitialStrainVector.resize(voigt_size, false);
```

Comment:
you should set the initial size of the vector before resizing. If it has already the correct size it will be much faster

Reply:
You cannot know the initial size because when you create the object the dimnesion is unknown..